compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1929,4 +1929,16 @@ object messages {
     val explanation =
       hl"An ${"object"} cannot have the same name as an existing ${"package"}. Rename either one of them."
   }
+
+  case class UnapplyInvalidNumberOfArguments(qual: untpd.Tree, argTypes: List[Type])(implicit ctx: Context)
+    extends Message(UnapplyInvalidNumberOfArgumentsID) {
+    val kind = "Syntax"
+    val msg = hl"Wrong number of argument patterns for $qual; expected: ($argTypes%, %)"
+    val explanation =
+      hl"""The Unapply method of $qual was used with incorrect number of arguments.
+          |Expected usage would be something like:
+          |case $qual(${argTypes.map(_ => '_')}%, %) => ...
+          |
+        |where subsequent arguments would have following types: ($argTypes%, %).""".stripMargin
```

Comment:
`""".stripMargin` should be on the next line.

Reply:
The issue with `stripMargin` being on the next line is that it adds an extra line to the explanation which is not desired.

Reply:
@nicolasstucki @allanrenucci can you please decide which option is recommended? I will gladly oblige.
Also - whoever does the merge can fix it as I have checked 'Allow edits from maintainers'.

Reply:
Use Allan's.